tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+	Expect(err).ToNot(HaveOccurred())
+
+	old, err := json.Marshal(ns)
+	Expect(err).ToNot(HaveOccurred())
+	new := ns.DeepCopy()
+	if new.Labels == nil {
+		new.Labels = map[string]string{}
+	}
+	new.Labels[key] = value
+
+	newJson, err := json.Marshal(new)
+	Expect(err).ToNot(HaveOccurred())
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = virtCli.CoreV1().Namespaces().Patch(context.Background(), ns.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func newServiceWithAppSelector(name, namespace, selector string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(serviceAccount, appLabel, istioProxyAnnotationValue, userData string, network *v1.Network, iface *v1.Interface) *v1.VirtualMachineInstance {
+	vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+	tests.AddServiceAccountDisk(vmi, serviceAccount)
+	tests.AddUserData(vmi, "disk1", userData)
+	addInjectIstioProxyAnnotation(vmi, istioProxyAnnotationValue)
+	addAppLabel(vmi, appLabel)
+	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*iface}
+	vmi.Spec.Networks = []v1.Network{*network}
+	return vmi
+}
+
+var _ = FSIGDescribe("[Serial] Istio", func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+	var vmiWithExplicitPorts *v1.VirtualMachineInstance
+	var vmiWithDefaultPorts *v1.VirtualMachineInstance
+
+	defaultNetwork := v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+	defaultInterface := v1.Interface{
+		Name: "default",
+		InterfaceBindingMethod: v1.InterfaceBindingMethod{
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+	}
+	interfaceWithExplicitPorts := v1.Interface{
+		Name: "default",
+		InterfaceBindingMethod: v1.InterfaceBindingMethod{
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+		Ports: []v1.Port{
+			{
+				Port: svcDeclaredTestPort,
+			},
+			{
+				Port: svcUndeclaredTestPort,
+			},
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			vmiWithExplicitPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
```

Comment:
Maybe export  the vm creation to an helper? 
The two vm objects creations are almost a duplication.

Reply:
I put some of the helper methods to utils, but not all of them as IMO some these are specific for istio scenario and it's unlikely that it would be used elsewhere.